var/spack/repos/builtin/packages/ipopt/package.py

```diff
@@ -68,6 +78,20 @@ def install(self, spec, prefix):
             "--with-lapack-lib=%s" % lapack_lib
         ]
 
+        if 'coinhsl' in spec:
+            coinhsl_dir = spec['coinhsl'].prefix
+            coinhsl_lib = '-L%s -lcoinhsl' % coinhsl_dir.lib
```

Comment:
the default implementation of `libs` look for `l<packagename>` library, which is your case.